### PR TITLE
[vcpkg-cmake] Search for binaries after manifest-based installs

### LINKED ISSUE #16604: [vcpkg.cmake] Binaries are missing in many manifest-based installs
**Describe the bug**
When using manifest-based installs in a project, `find_package()` does not find any associated binaries in named dependencies.

**Environment**
- OS: Any
- Compiler: Any

**To Reproduce**
1. Define a vcpkg.json such as:
  ```
{
    "name": "bin-fail",
    "version": "1.0.0",
    "dependencies": [
        "protobuf"
    ]
}
```
2. Create a CMakeFiles.txt that has the following:
```
find_package(Protobuf CONFIG REQUIRED)
```
3. Run cmake and observe in CMakeCache.txt the following line:
```
//The Google Protocol Buffers Compiler
Protobuf_PROTOC_EXECUTABLE:FILEPATH=Protobuf_PROTOC_EXECUTABLE-NOTFOUND
```

**Expected behavior**
I expect `find_package()` to find the executable.

### PR DESCRIPTION
**Pull request by townsend2010**
Otherwise, find_package() fails to find any binaries for many manifest installed packages until a second cmake run commences.

**Describe the pull request**

- What does your PR fix? Fixes #16604

- Which triplets are supported/not supported? Have you updated the CI baseline? N/A

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Seems so:)

### CODE REVIEW

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -655,6 +646,15 @@ if(VCPKG_MANIFEST_MODE AND VCPKG_MANIFEST_INSTALL AND NOT Z_VCPKG_CMAKE_IN_TRY_C
...
 
+list(APPEND CMAKE_PROGRAM_PATH "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/tools")
+
+file(GLOB Z_VCPKG_TOOLS_DIRS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/tools/*")
+foreach(Z_VCPKG_TOOLS_DIR IN LISTS Z_VCPKG_TOOLS_DIRS)
+    if(IS_DIRECTORY "${Z_VCPKG_TOOLS_DIR}")
+        list(APPEND CMAKE_PROGRAM_PATH "${Z_VCPKG_TOOLS_DIR}")
+    endif()
+endforeach()
+
```
Comment (Neumann-A) [inline scripts/buildsystems/vcpkg.cmake:657]: Could you point out the difference? (except for whitespace changes)
I am either blind or there is no relevant change?

Comment (townsend2010) [reply scripts/buildsystems/vcpkg.cmake]: Oops, the whitespace change was cruft and have fixed that.

But to answer your question, tt's moving the block that searches for for `tools` **after** manifest mode install block.

**Review by strega-nil — APPROVED** [review]

**Comment by townsend2010** [comment]
Great, thanks!

**Comment by strega-nil** [comment]
thank you!

**Pull request merged**